prow/spyglass/spyglass.go

```diff
@@ -248,9 +248,12 @@ func (s *Spyglass) RunToPR(src string) (string, string, int, error) {
 		if logType == gcs.NonPRLogs {
 			return "", "", 0, fmt.Errorf("not a PR URL: %q", key)
 		} else if logType == gcs.PRLogs {
+			if len(split) < 3 {
+				return "", "", 0, fmt.Errorf("malformed %s key %q should have at least three components", gcs.PRLogs, key)
+			}
 			prNum, err := strconv.Atoi(split[len(split)-3])
 			if err != nil {
-				return "", "", 0, fmt.Errorf("couldn't parse PR number %q in %q: %v", split[5], key, err)
+				return "", "", 0, fmt.Errorf("couldn't parse PR number %q in %q: %v", split[len(split)-3], key, err)
```

Comment:
Can we use a variable for `split[len(split)-3]` to DRY out that expression in this stanza?

Reply:
Probably a good idea, given getting it wrong is the origin of this bug.